lesson16/main.js

```diff
@@ -0,0 +1,117 @@
+   const fragment = document.createDocumentFragment();
+   for (let i = 0; i < values.length; i++) {
+      const categoryTabItem = createElementWithClassName("li", "category-tab__item")
+      categoryTabItem.id = `js-categoryTabItem${i}`;
+      const categoryTabText = createElementWithClassName("a", "category-tab__text");
+      categoryTabText.textContent = values[i].category
+      fragment.appendChild(categoryTabItem).appendChild(categoryTabText);
+   }
+   categoryTab.appendChild(fragment);
+   //初期表示のタブの選択はデータとして持っているがまだ理解できていないので、現段階ではNewsのタブとしてある
+   const newsTabItem = document.getElementById("js-categoryTabItem0");
+   newsTabItem.classList.add("is-active");
+}
+
+const renderTabContainer = (values) => {
+   const tabFragment = document.createDocumentFragment();
+   const tabContainer = createElementWithClassName("div", "tab-container");
+   tabContainer.id = "js-tabContainer";
+   for (let i = 0; i < values.length; i++) {
+      const tabContents = createElementWithClassName("div", "tab-contents");
+      tabContents.id = `js-tabContents${i}`;
+      const tabLists = createElementWithClassName("ul", "tab-contents__lists");
+      tabLists.id = `js-tabLists${i}`;
+      const tabImage = createElementWithClassName("img", "tab-contents__image");
+      tabImage.id = `js-tabImage${i}`;
+      tabImage.src = values[i].img;
+      tabContents.appendChild(tabLists).appendChild(createArticlesTitles(values[i]));
+      tabFragment.appendChild(tabContainer).appendChild(tabContents).appendChild(tabImage);
+   }
+   body.insertBefore(tabFragment, categoryTab.nextElementSibling);
+   //TODO:どのカテゴリタブを初期表示時に選んでいるかはデータとして持っている
+   //一旦、Newカテゴリーを初期表示としている
+   const newsContents = document.getElementById("js-tabContents0");
+   newsContents.classList.add("is-show");
+}
+
+const createArticlesTitles = (values) => {
+   const articlesTitlesFragment = document.createDocumentFragment();
+   for (let i = 0; i < values.articles.length; i++) {
+      const articlesHref = values.articles.map((value) => value.href);
+      const articlesTitles = values.articles.map((value) => value.title);
+      const tabItems = createElementWithClassName("li", "lists__item");
+      const tabItemsHref = createElementWithClassName("a", "lists__item__text");
+      tabItemsHref.textContent = articlesTitles[i];
+      tabItemsHref.href = articlesHref[i];
+      articlesTitlesFragment.appendChild(tabItems).appendChild(tabItemsHref);
+   }
+   return articlesTitlesFragment;
+}
+
+   categoryTab.addEventListener("click", (e) => {
+      const activeTab = document.querySelector(".is-active");
+      const showContents = document.querySelector(".is-show");
+      activeTab.classList.remove("is-active");
+      e.target.classList.add("is-active");
+      const arrayCategoryTab = [...document.querySelectorAll(".category-tab__item")]
+      const isActive = document.getElementsByClassName("is-active")[0];
+      const index = arrayCategoryTab.indexOf(isActive);
```

Comment:
IMO
It is also possible to get the index of the clicked element in this way
```suggestion
      const index = arrayCategoryTab.indexOf(e.target);
```

Reply:
Thank you for  your advise.
I hadn't noticed. 
Fixed as suggested.
[4a1eb6b](https://github.com/taka0209/js_lesson/pull/18/commits/4a1eb6b04a8710f9d9c8277d67c261fa521af574)